packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/search/PaletteSearchField.tsx

```diff
@@ -47,26 +47,18 @@ export const PaletteSearchField = ({ onValueChanged, onEscape, onDirectEditClick
   const onTextClear = (): void => {
     setState((prevState) => ({ ...prevState, value: '' }));
     onValueChanged('');
-  };
-
-  const handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>): void => {
-    const { key } = event;
-    if (key === 'Escape' && onEscape) {
-      onEscape();
-    } else if (key === 'F2') {
-      onDirectEditClick();
-    }
+    document.getElementById('palette-search-field')?.focus();
```

Comment:
Leverage useRef instead, this is more in line with the react way of doing things.

Reply:
Done